### PR TITLE
Fix/update progression

### PR DESCRIPTION
**Pull request by Volubyl**
## Detailed purpose of the PR

This PR does :

- sort the progression chronoligcally before sending to the server
- display the no content modal only in one case : when the platform is really disabled (ignore other 403 http headers case)

Next progression behavior :

After reached the final level and clicked on the "back home button", all the stored progressions will be sent to the server. Few progressions could be rejected by the server (403 error) but once the user will return to the home page all of the remaining progression will be sync successfully.

__Remains to do__ : 

- fix API progression 

### Testing strategy

- [X] I ensured my code is tested:  UT and always with ❤️ 
- [ ] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

File: src/utils/progressions.js [hunk 1]
```diff
@@ -18,3 +19,6 @@ export const isFailure = (progression: Progression) => {
 export const isDone = (progression: Progression) => {
   return isFailure(progression) || isSuccess(progression);
 };
+
+export const sortProgressionChronologicaly = (progressions: Progression[]): Progression[] =>
```
Comment (Volubyl) [inline src/utils/progressions.js]: a fresh progression is a progression containing the newer action performed by the user. By definition the newest action is the last action stored (index 0)

File: src/__fixtures__/progression.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
 // @flow
-import type {GenericContent, Progression, State} from '@coorpacademy/progression-engine';
+import type {GenericContent, Progression, State, Action} from '@coorpacademy/progression-engine';
 import type {Engine} from '../types';
 
 export type StateExtension = $Shape<State>;
 
+export const createAction = ({createdAt}: {createdAt?: string}): Action => {
```
Comment (VincentCATILLON) [inline src/__fixtures__/progression.js]: 👍

File: src/utils/progressions.js (hunk 1 above)
Comment (VincentCATILLON) [reply src/utils/progressions.js]: `Array<Progression>` please

File: src/utils/progressions.js [hunk 3]
```diff
@@ -18,3 +19,6 @@ export const isFailure = (progression: Progression) => {
...
+  orderBy(['actions[0].createdAt'], ['asc'], progressions);
```
Comment (VincentCATILLON) [inline src/utils/progressions.js]: :/ 

In ES6 (and covered by flow): 
```
progressions.sort((a: Progression, b: Progression) => {
  const aDate = a.actions[0] && a.actions[0].createdAt;
  const bDate = b.actions[0] && b.actions[0].createdAt;

  return aDate - bDate;
});
```

File: src/modules/fetch.js [hunk 4]
```diff
@@ -4,14 +4,30 @@ import _fetch from 'cross-fetch';
...
-const fetch: typeof _fetch = async (...args) => {
-  const result = await _fetch(...args);
+export const ERROR_MESSAGE = 'The "mobile" feature is currently unavailable on your platform';
 
-  if (result && result.status === 403) {
-    throw new ForbiddenError('Fetch Forbidden');
+const fetch: typeof _fetch = async (url, options) => {
+  const headers = options && options.headers;
+
+  const newHeaders = {
```
Comment (AdrieanKhisbe) [inline src/modules/fetch.js]: idealement un lodash/fp.set aurait suffit. 
et comme ça on aurait meme pas à se casser la tete pour nommer la variable (en effet newHeaders c'est pas ouf)

File: src/modules/fetch.js [hunk 5]
```diff
@@ -4,14 +4,30 @@ import _fetch from 'cross-fetch';
...
+    ...headers,
+    'X-Requested-With': 'XMLHttpRequest'
```
Comment (Fausto95) [inline src/modules/fetch.js]: what is this?

File: src/modules/fetch.js [hunk 6]
```diff
@@ -4,14 +4,30 @@ import _fetch from 'cross-fetch';
 
 import {ForbiddenError} from '../models/error';
 
-const fetch: typeof _fetch = async (...args) => {
-  const result = await _fetch(...args);
+export const ERROR_MESSAGE = 'The "mobile" feature is currently unavailable on your platform';
```
Comment (AdrieanKhisbe) [inline src/modules/fetch.js]: c'est pas un peu fragile ça?

**Comment by AdrieanKhisbe** [comment]
> Array<Progression> please
@VincentCATILLON  why?

File: src/utils/progressions.js [hunk 7]
```diff
@@ -23,7 +23,14 @@ export const sortProgressionChronologicaly = (
   progressions: Array<Progression>
 ): Array<Progression> =>
   progressions.sort((a: Progression, b: Progression) => {
-    const aDate = a.actions[0].createdAt;
-    const bDate = b.actions[0].createdAt;
-    return aDate > bDate;
+    if (a.actions && a.actions.length > 0 && b.actions && b.actions.length > 0) {
```
Comment (Fausto95) [inline src/utils/progressions.js]: This is tooooo weird tbh

**Pull request merged**